Modules/Core/Common/include/itkImageRegion.h

```diff
@@ -283,28 +283,20 @@ class ITK_TEMPLATE_EXPORT ImageRegion final : public Region
     return true;
   }
 
-  /** Test if a region (the argument) is completely inside of this region. If
-   * the region that is passed as argument to this method, has a size of value
-   * zero, then it will not be considered to be inside of the current region,
-   * even its starting index is inside. */
+  /** Test if a region (the argument) is completely inside of this region. */
   bool
-  IsInside(const Self & region) const
+  IsInside(const Self & otherRegion) const
   {
-    IndexType beginCorner = region.GetIndex();
+    const auto otherIndex = otherRegion.m_Index;
+    const auto otherSize = otherRegion.m_Size;
 
-    if (!this->IsInside(beginCorner))
-    {
-      return false;
-    }
-    IndexType        endCorner;
-    const SizeType & size = region.GetSize();
     for (unsigned int i = 0; i < ImageDimension; ++i)
     {
-      endCorner[i] = beginCorner[i] + static_cast<OffsetValueType>(size[i]) - 1;
-    }
-    if (!this->IsInside(endCorner))
-    {
-      return false;
+      if ((otherIndex[i] < m_Index[i]) || ((otherIndex[i] + static_cast<IndexValueType>(otherSize[i])) >
+                                           (m_Index[i] + static_cast<IndexValueType>(m_Size[i]))))
```

Comment:
`+` has higher operator precedence than '<` and '>'; and `<` and `>` have higher precedence than `||`. So some parentheses can be removed here.  (Unless this is to influence `clang-format`??)

Please don't slow down the merge just because of this trivia.

Reply:
@Leengit Thanks for your suggestion. I just placed those extra parentheses there for clarity. But I'm willing to remove them if it is the preferred coding style nowadays. Does anyone else have an opinion on them?

Reply:
Now that Lee mentioned it, I agree that there could be less parentheses.

Reply:
> I agree that there could be less parentheses.

How many do you want?  😸

Reply:
@Leengit @dzenanz Now I see at https://github.com/InsightSoftwareConsortium/ITKSoftwareGuide/blob/36b553127ead40247889028306fc2fd566aa1e65/SoftwareGuide/Latex/Appendices/CodingStyleGuide.tex#L2203-L2209 

```
Unnecessary parentheses for grouping hinder reading the code.
The C++ precedence and associativity (the order in which the operands are
evaluated) of operators must be taken into account to avoid using unnecessary
parentheses.

As a general principle, these apply to condition expressions, and statements
where mathematical, logical or bitwise operations are performed.
```

C++ Core Guidelines (January 3, 2022) appears to encourage the use of parentheses only _slightly_ more (specifically for _bitwise_ logical operations): "If in doubt about operator precedence, parenthesize" https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-parens

OK, so (from now) I will (try to) change my habit regarding this style (by reducing the number of parentheses (that I will be using)).  😺

Reply:
Just to document: The bitwise operators (`&`, `|`, ...), like the logical operators (`&&`, `||`), have an operator precedence lower than the relational operators (`<`, `==`, ...)  and can be used for logical operations that don't need short circuiting.  However when the intent, like with arithmetic (`+`, `-`, ...), is to have the precedence be higher than the relational operators then parentheses are usually required.  I try to avoid the parentheses in the first use case, though sometimes people will use them anyway to make sure that folks are not confused with the precedence that would be appropriate for the second use case.

Reply:
OK! Thanks! Just removed those parentheses that aren't strictly necessary (and also did `git rebase master`), with my latest [force-push](https://github.com/InsightSoftwareConsortium/ITK/compare/165e9e74e49393eebb3001e91e770d8641589d8c..25ee2e960d70b8a7a2a133400d8b815605d6707b)